servant-auth-server/src/Servant/Auth/Server.hs

```diff
@@ -1,160 +1,171 @@
-  -- server (or by some other party posessing the relevant key), and used to
-  -- indicate the bearer's identity or authorization.
-  --
-  -- Arbitrary information can be encoded - just declare instances for the
-  -- 'FromJWT' and 'ToJWT' classes. Don't go overboard though - be aware that
-  -- usually you'll be trasmitting this information on each request (and
-  -- response!).
-  --
-  -- Note that, while the tokens are signed, they are not encrypted. Do not put
-  -- any information you do not wish the client to know in them!
-
-  -- ** Combinator
-  -- | Re-exported from 'servant-auth'
-  , JWT
-
-  -- ** Classes
-  , FromJWT(..)
-  , ToJWT(..)
-
-  -- ** Related types
-  , IsMatch(..)
-
-  -- ** Settings
-  , JWTSettings(..)
-  , defaultJWTSettings
-
-  -- ** Create check
-  , jwtAuthCheck
-
-
-  ----------------------------------------------------------------------------
-  -- * Cookie
-  -- | Cookies are also a method of identifying and authenticating a user. They
-  -- are particular common when the client is a browser
-
-  -- ** Combinator
-  -- | Re-exported from 'servant-auth'
-  , Cookie
-
-  -- ** Settings
-  , CookieSettings(..)
-  , XsrfCookieSettings(..)
-  , defaultCookieSettings
-  , defaultXsrfCookieSettings
-  , makeSessionCookie
-  , makeSessionCookieBS
-  , makeXsrfCookie
-  , makeCsrfCookie
-  , makeCookie
-  , makeCookieBS
-  , acceptLogin
-  , clearSession
-
-
-  -- ** Related types
-  , IsSecure(..)
-  , SameSite(..)
-  , AreAuths
-
-  ----------------------------------------------------------------------------
-  -- * BasicAuth
-  -- ** Combinator
-  -- | Re-exported from 'servant-auth'
-  , BasicAuth
-
-  -- ** Classes
-  , FromBasicAuthData(..)
-
-  -- ** Settings
-  , BasicAuthCfg
-
-  -- ** Related types
-  , BasicAuthData(..)
-  , IsPasswordCorrect(..)
-
-  -- ** Authentication request
-  , wwwAuthenticatedErr
-
-  ----------------------------------------------------------------------------
-  -- * Utilies
-  , ThrowAll(throwAll)
-  , generateKey
-  , generateSecret
-  , fromSecret
-  , writeKey
-  , readKey
-  , makeJWT
-
-  -- ** Re-exports
-  , Default(def)
-  , SetCookie
-  ) where
-
-import Prelude hiding                           (readFile, writeFile)
-import Data.ByteString                          (ByteString, writeFile, readFile)
-import Data.Default.Class                       (Default (def))
+  ( -- | This package provides implementations for some common authentication
+    -- methods. Authentication yields a trustworthy (because generated by the
+    -- server) value of an some arbitrary type:
+    --
+    -- > type MyApi = Protected
+    -- >
+    -- > type Protected = Auth '[JWT, Cookie] User :> Get '[JSON] UserAccountDetails
+    -- >
+    -- > server :: Server Protected
+    -- > server (Authenticated usr) = ... -- here we know the client really is
+    -- >                                  -- who she claims to be
+    -- > server _ = throwAll err401
```

Comment:
Shouldn't this be rewritten?

Reply:
Yup. throwAll should also be removed, and servant-auth-client tests fixed. I hopefully will get this done today.

Reply:
Done with these.